olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -164,6 +169,152 @@ client::CancellableFuture<DataResponse> VersionedLayerClientImpl::GetData(
                                                  std::move(promise));
 }
 
+client::CancellationToken VersionedLayerClientImpl::PrefetchPartitions(
+    PartitionsRequest request, PrefetchPartitionsResponseCallback callback,
+    PrefetchPartitionsStatusCallback status_callback) {
+  using EmptyResponse = Response<client::ApiNoResult>;
+  using client::CancellationContext;
+  using client::ErrorCode;
+  auto catalog = catalog_;
+  auto layer_id = layer_id_;
+  auto lookup_client = lookup_client_;
+  // shared_settings for saving some mb while prefetch.
+  auto shared_settings = std::make_shared<client::OlpClientSettings>(settings_);
+
+  using PartitionDataHandleResult = std::map<std::string, std::string>;
```

Comment:
Once again, why do you need a map? Do you need partitions to be sorted or why take into account such extra overhead?